polar-core/tests/integration_tests.rs

```diff
@@ -1535,3 +1575,65 @@ fn test_list_results() {
     assert!(qeval(&mut polar, "xs = [2] and [1,2] = [1, *xs]"));
     assert!(qnull(&mut polar, "[1, 2] = [2, *ys]"));
 }
+
+#[test]
+fn test_expressions_in_lists() {
+    let mut polar = Polar::new();
+    let policy = r#"
+    scope(actor: Dictionary, "read", "Person", filters) if
+        filters = ["id", "=", actor.id];
+    "#;
+    polar.load(policy).unwrap();
+    assert!(qeval(
+        &mut polar,
+        r#"scope({id: 1}, "read", "Person", ["id", "=", 1])"#
+    ));
+    assert!(qnull(
+        &mut polar,
+        r#"scope({id: 2}, "read", "Person", ["id", "=", 1])"#
+    ));
+    assert!(qnull(
+        &mut polar,
+        r#"scope({id: 1}, "read", "Person", ["not_id", "=", 1])"#
+    ));
+    assert!(qeval(&mut polar, r#"d = {x: 1} and [d.x, 1+1] = [1, 2]"#));
+    assert_eq!(
+        qvar(
+            &mut polar,
+            r#"d = {x: 1} and [d.x, 1+1] = [1, *rest]"#,
+            "rest"
+        ),
+        vec![value!([value!(2)])]
+    );
+}
+
+#[test]
+fn test_list_matches() {
+    let mut polar = Polar::new();
+
+    assert!(qeval(&mut polar, "[] matches []"));
+    assert!(qnull(&mut polar, "[1] matches []"));
+    assert!(qnull(&mut polar, "[] matches [1]"));
+    assert!(qnull(&mut polar, "[1, 2, 3] matches [1, 2]"));
+    assert!(qnull(&mut polar, "[1, 2] matches [1, 2, 3]"));
+    assert!(qnull(&mut polar, "[2, 1] matches [1, 2]"));
+    assert!(qeval(&mut polar, "[1, 2, 3] matches [1, 2, 3]"));
+    assert!(qnull(&mut polar, "[1, 2, 3] matches [1, 2]"));
+
+    assert!(qeval(&mut polar, "[] matches []"));
+    assert!(qnull(&mut polar, "[x] matches []"));
+    assert!(qnull(&mut polar, "[] matches [x]"));
+    assert!(qnull(&mut polar, "[1, 2, x] matches [1, 2]"));
+    assert!(qnull(&mut polar, "[1, x] matches [1, 2, 3]"));
+    assert!(qnull(&mut polar, "[2, x] matches [1, 2]"));
+    assert!(qeval(&mut polar, "[1, 2, x] matches [1, 2, 3]"));
```

Comment:
Should we use `qvar` here to test that `x` gets bound to 3?

Reply:
Yes